src/Psl/Type/Internal/ShapeType.php

```diff
@@ -51,6 +52,10 @@ public function __construct(
      */
     public function coerce(mixed $value): array
     {
+        if ($value instanceof stdClass) {
```

Comment:
Question: perhaps `get_class($value) === stdClass::class` too? :thinking: 

I don't know if people are crazy enough to extend `stdClass` in 2023, but given that the PHP community has regressed to architectural styles from 2007 over time...

Reply:
Even if people are crazy enough to extend `stdClass` in 2024 😉 , it coult still be coerced into the specified shape.
So I don't see a big issue with just using the `instanceof` check.

Unless you have a different opinion, this one looks good to go to me!